redis.c

```diff
@@ -2066,6 +2069,15 @@ redis_sock_read_multibulk_multi_reply_loop(INTERNAL_FUNCTION_PARAMETERS,
                                            RedisSock *redis_sock, zval *z_tab)
 {
     fold_item *fi;
+    int count = 0;
+
+    /* Count issued commands so we can preallocate array for results */
+    for (fi = redis_sock->head; fi; /* void */) {
+        count++;
+        if (fi) fi = fi->next;
+    }
+
+    array_init_size(z_tab, count);
```

Comment:
It would be interesting to profile whether this is actually faster in practice, given that the `HashTable` grows exponentially.